include/ardpolyfills/arduino.h

```diff
@@ -16,20 +16,19 @@
  *
  */
 
- #ifndef Arduino_h
+#ifndef Arduino_h
 #define Arduino_h
 #include <cctype>
 #include <cmath>
 #include <cstdlib>
-#include <ctype.h>
 
 // Arduino Math functions
 using std::abs;
 long constrain(long, long, long);
 long map(long, long, long, long, long);
-long min(long, long);
-long max(long, long);
-long sq(long);
+template <typename T> T min(T, T);
+template <typename T> T max(T, T);
+template <typename T> T sq(T);
```

Comment:
Template definitions must be in a header
Also, why not use `std::min` and `std::max`?

Reply:
ok, maybe I should have used the min & max.